src/agent/rustjail/src/process.rs

```diff
@@ -93,7 +93,7 @@ impl ProcessOperations for Process {
     }
 
     fn signal(&self, sig: Signal) -> Result<()> {
-        signal::kill(self.pid(), Some(sig))
+        signal::kill(Pid::from_raw(-self.pid), Some(sig))
```

Comment:
probably need to separate the signal and only do this when it's SIGTERM/SIGKILL. Will update tomorrow.